oauth2_http/javatests/com/google/auth/oauth2/DefaultCredentialsProviderTest.java

```diff
@@ -326,6 +328,32 @@ public void getDefaultCredentials_envUser_providesToken() throws IOException {
     testUserProvidesToken(testProvider, USER_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN);
   }
 
+  @Test
+  public void getDefaultCredentials_quota_project() throws IOException {
+    InputStream userStream =
+        UserCredentialsTest.writeUserStream(
+            USER_CLIENT_ID, USER_CLIENT_SECRET, REFRESH_TOKEN, QUOTA_PROJECT);
+    TestDefaultCredentialsProvider testProvider = new TestDefaultCredentialsProvider();
+    String userPath = tempFilePath("user.json");
+    testProvider.addFile(userPath, userStream);
+    testProvider.setEnv(DefaultCredentialsProvider.CREDENTIAL_ENV_VAR, userPath);
+    MockTokenServerTransportFactory transportFactory = new MockTokenServerTransportFactory();
+    transportFactory.transport.addClient(USER_CLIENT_ID, USER_CLIENT_SECRET);
+    transportFactory.transport.addRefreshToken(REFRESH_TOKEN, ACCESS_TOKEN);
+
+    // Validate that quota from env overrides the value from json
+    testProvider.setEnv(
+        DefaultCredentialsProvider.QUOTA_PROJECT_ENV_VAR, QUOTA_PROJECT_FROM_ENVIRONMENT);
+    GoogleCredentials credentials = testProvider.getDefaultCredentials(transportFactory);
+    assertEquals(QUOTA_PROJECT_FROM_ENVIRONMENT, credentials.getQuotaProjectId());
+
+    // Validate that if user sets quota, env and json value not used
+    credentials = credentials.toBuilder().setQuotaProjectId(QUOTA_PROJECT_EXPLICIT).build();
```

Comment:
Add a test case where: 
- env variable for quota is set
- user creates credential from file himself, the file contains quota (ideally)
- check quota is not affected by env variable

Reply:
createWithQuotaProject method needs to be called explicitly to set quota from env. With current implementation we need a environment provider to be sent explicitly for using custom env source.
Since currently there is no way to pass in the environmentprovider while class initialization, an effective test is not possible.

If we can use a mocking library instead like mentioned in https://stackoverflow.com/questions/8168884/how-to-test-code-dependent-on-environment-variables-using-junit, then such tests will be easy.
We can achieve same by adding EnvironmentProvider as a property/field in GoogleCredentials but then there will be a complicated change to add it to all builders, constructors etc. Not sure it is worth doing that.

Reply:
Added Integration test for this as discussed